Demo/Common/IntegrationExamples/AdvancedFlowExample.swift

```diff
@@ -0,0 +1,374 @@
+        do {
+            try (advancedFlowComponent as? DropInComponent)?.reload(with: order, paymentMethods)
+        } catch {
+            finish(with: error)
+        }
+    }
+
+    internal func finish(with result: PaymentsResponse) {
+        let success = result.resultCode == .authorised || result.resultCode == .received || result.resultCode == .pending
+        let message = "\(result.resultCode.rawValue) \(result.amount?.formatted ?? "")"
+        finalize(success, message)
+    }
+
+    internal func finish(with error: Error) {
+        let message: String
+        if let componentError = (error as? ComponentError), componentError == ComponentError.cancelled {
+            message = "Cancelled"
+        } else {
+            message = error.localizedDescription
+        }
+        finalize(false, message)
+    }
+
+    private func finalize(_ success: Bool, _ message: String) {
+        advancedFlowComponent?.finalizeIfNeeded(with: success) { [weak self] in
+            guard let self = self else { return }
+            self.dismissAndShowAlert(success, message)
+        }
+    }
+
+    internal func dismissAndShowAlert(_ success: Bool, _ message: String) {
+        presenter?.dismiss {
+            // Payment is processed. Add your code here.
+            let title = success ? "Success" : "Error"
+            self.presenter?.presentAlert(withTitle: title, message: message)
+        }
+    }
+
+}
+extension AdvancedFlowExample: DropInComponentDelegate {
+
+    func didSubmit(_ data: PaymentComponentData, from component: PaymentComponent, in dropInComponent: AnyDropInComponent) {
+        didSubmit(data, from: component)
+    }
+
+    func didFail(with error: Error, from component: PaymentComponent, in dropInComponent: AnyDropInComponent) {
+        didFail(with: error, from: component)
+    }
+
+    func didProvide(_ data: ActionComponentData, from component: ActionComponent, in dropInComponent: AnyDropInComponent) {
+        didProvide(data, from: component)
+    }
+
+    func didComplete(from component: ActionComponent, in dropInComponent: AnyDropInComponent) {
+        didComplete(from: component)
+    }
+
+    func didFail(with error: Error, from component: ActionComponent, in dropInComponent: AnyDropInComponent) {
+        didFail(with: error, from: component)
+    }
+
+    internal func didCancel(component: PaymentComponent, from dropInComponent: AnyDropInComponent) {
+        // Handle the event when the user closes a PresentableComponent.
+        print("User did close: \(component.paymentMethod.name)")
+    }
+
+    internal func didFail(with error: Error, from dropInComponent: AnyDropInComponent) {
+        finish(with: error)
+    }
+
+}
+
+extension AdvancedFlowExample: PaymentComponentDelegate {
+
+    internal func didSubmit(_ data: PaymentComponentData, from component: PaymentComponent) {
+        let request = PaymentsRequest(data: data)
+        apiClient.perform(request, completionHandler: paymentResponseHandler)
```

Comment:
This captures self since `paymentResponseHandler` is a function of `self`, same as currently. let's change it.

Reply:
Yes you are right. I have made the change.da858bb9a858774fa139005a65bb215f94ed0c10